Userland/Libraries/LibSemVer/SemVer.cpp

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright (c) 2023, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <AK/Find.h>
+#include <AK/Function.h>
+#include <AK/GenericLexer.h>
+#include <AK/ReverseIterator.h>
+#include <AK/StringBuilder.h>
+#include <LibRegex/Regex.h>
+#include <LibRegex/RegexParser.h>
+#include <LibSemVer/SemVer.h>
+
+namespace SemVer {
+String SemVer::suffix() const
+{
+    StringBuilder sb;
+    if (m_prerelease_identifiers.size() > 0)
+        sb.appendff("-{}", prerelease());
+    if (m_buildmetadata.code_points().length() > 0)
+        sb.appendff("+{}", buildmetadata());
+    return MUST(sb.to_string());
```

Comment:
Not sure if we should consider this `.release_value_but_fixme_should_propagate_errors()`, since someone could potentially craft arbitrarily long strings with this.

Reply:
I see, it has VERIFY macro in it. But what's the coherence `but_fix_me` make (in the function name) with the definition of it.

Reply:
`MUST` also has a `VERIFY` in it. The difference between `MUST` and `release_value_but_fixme_should_propagate_errors` is purely on the documentation side, as the former is for cases where we _know_ it can't/shouldn't fail, while the latter is for cases where it _can_ fail but we don't have everything setup to handle/propagate the resulting error.

The reason why I'm unsure is because we have since decided that very small string allocations are okay to assume that they never fail, but I don't know whether this extends to cases like this (where one could format a very long string still).

Reply:
Ah, documentation side. That makes sense, one of the case if OOM then we have reached an impasse.